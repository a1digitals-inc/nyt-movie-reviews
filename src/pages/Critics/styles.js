import styled from 'styled-components';
import { darken } from 'polished';

export const Container = styled.div`
  max-width: 1020px;
  margin: 0 auto;
  padding: 0 20px 50px;
  margin-top: 15px;
`;

export const CriticList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  flex-direction: row;
  justify-content: center;
  text-align: center;

  li {
    width: 250px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 20px;
    margin: 0 15px 15px 0;
    background: #fff;
    border: 1px solid ${darken(0.2, '#e6e6e6')};
    border-radius: 4px;

    &:hover {
      background: ${darken(0.04, '#fff')};
    }

    div {
      max-height: 250px;
      overflow: hidden;
    }

    strong {
      margin-top: 15px;
      font-size: 20px;
      color: #000000;
    }

    small {
      margin-top: 15px;
      color: #999;
      max-height: 100px;
      overflow-x: scroll;
      overflow-x: hidden;
    }

    a {
      margin-top: 15px;
      color: #999;
      font-weight: bold;

      &:hover {
        color: ${darken(0.5, '#999')};
      }
    }
  }
`;
