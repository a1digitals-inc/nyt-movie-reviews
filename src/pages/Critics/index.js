import { toast } from 'react-toastify';
import React, { useState, useEffect } from 'react';

import api from '../../services/api';
import apiConfig from '../../config/api';

import PageLoading from '../../components/PageLoading';
import NotFound from '../../components/NotFound';

import { Container, CriticList } from './styles';

import emptyCriticLogo from '../../assets/images/empty-critic.svg';

export default function Critics() {
  const [isLoading, setIsLoading] = useState(true);
  const [critics, setCritics] = useState([]);

  useEffect(() => {
    async function loadCritics() {
      try {
        const response = await api.get(
          `critics/all.json?api-key=${apiConfig.key}`
        );

        setCritics(response.data.results);
        setIsLoading(false);
      } catch (err) {
        toast.error('Failed to load critics');
        setIsLoading(false);
      }
    }

    loadCritics();
  }, []);

  if (isLoading) {
    return <PageLoading />;
  }

  return (
    <>
      {!critics.length ? (
        <NotFound Message="No records found" />
      ) : (
        <Container>
          <CriticList>
            {critics.map(critic => {
              const image =
                critic.multimedia && critic.multimedia.resource
                  ? critic.multimedia.resource.src
                  : null;
              return (
                <li key={critic.display_name}>
                  <div>
                    <img src={image || emptyCriticLogo} alt="critic" />
                  </div>
                  <strong>{critic.display_name}</strong>
                  <small>{critic.bio}</small>
                  <a href={`/reviews/${critic.display_name}`}>
                    See movie reviews
                  </a>
                </li>
              );
            })}
          </CriticList>
        </Container>
      )}
    </>
  );
}
