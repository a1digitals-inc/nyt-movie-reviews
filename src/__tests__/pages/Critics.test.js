import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Critics from '../../pages/Critics';

describe('Critics page', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {
    critics: {
      loading: false,
      critics: [],
    },
  };
  const store = mockStore(initialState);

  it('should render without crashing', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Critics />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
