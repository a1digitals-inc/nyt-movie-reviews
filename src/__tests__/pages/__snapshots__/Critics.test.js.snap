// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Critics page should render without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Critics>
    <NotFound
      Code={null}
      Message="No records found"
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-ifAKCX",
                "isStatic": false,
                "lastClassName": "ioktai",
                "rules": Array [
                  "
  display: flex;
  justify-content: center;
  align-items: center;
  font-style: oblique;
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-ifAKCX",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="sc-ifAKCX ioktai"
          >
            <styled.div>
              <StyledComponent
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-EHOje",
                      "isStatic": false,
                      "lastClassName": "cTEuPM",
                      "rules": Array [
                        "
  display: flex;
  flex-direction: column;
  text-align: center;
  margin-top: 90px;
",
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-EHOje",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
              >
                <div
                  className="sc-EHOje cTEuPM"
                >
                  <strong />
                  <strong>
                    No records found
                  </strong>
                </div>
              </StyledComponent>
            </styled.div>
          </div>
        </StyledComponent>
      </styled.div>
    </NotFound>
  </Critics>
</Provider>
`;
